applications/asset_tracker_v2/tests/cloud_codec_nrf/src/nrf_cloud_codec_test.c

```diff
@@ -0,0 +1,686 @@
+		"\"accTimeoutInact\":1,"\
+		"\"nod\":["\
+			"\"gnss\","\
+			"\"ncell\""\
+		"]"\
+	"}"\
+"}"
+
+#define CONF_SEND_EXAMPLE "{\"state\":{\"reported\":" CONF_RECV_EXAMPLE "}}"
+
+#define UI_EXAMPLE \
+"{"\
+	"\"data\":\"1\","\
+	"\"appId\":\"BUTTON\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123"\
+"}"
+
+#define BAT_BATCH_EXAMPLE \
+"[{"\
+	"\"appId\":\"VOLTAGE\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123,"\
+	"\"data\":\"50\""\
+"}]"
+
+/** TODO: this differs from the schemas
+ * https://github.com/nRFCloud/application-protocols/blob/v1/schemas/deviceToCloud/gnss/GPGGA-example.json
+ */
+#define GNSS_BATCH_EXAMPLE \
+"[{"\
+	"\"appId\":\"GNSS\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123,"\
+	"\"data\":\"$GPGGA,141856.00,6326.27336,N,01028.08721,E,1,09,0.81,34.7,M,39.8,M,,*64\\n\""\
+"}]"
+
+/** TODO: this differs from the schema
+ * https://github.com/nRFCloud/application-protocols/blob/v1/schemas/deviceToCloud/device/device.json
+ */
+#define MODEM_STATIC_BATCH_EXAMPLE \
+"[{"\
+	"\"appId\":\"DEVICE\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123,"\
+	"\"data\":{"\
+		"\"deviceInfo\":{"\
+			"\"imei\":\"352656106111232\","\
+			"\"iccid\":\"89450421180216211234\","\
+			"\"modemFirmware\":\"mfw_nrf9160_1.2.3\","\
+			"\"board\":\"nrf9160dk_nrf9160\","\
+			"\"appVersion\":\"v1.0.0-development\""\
+		"}"\
+	"}"\
+"}]"
+
+/** TODO: this differs from the schema
+ * https://github.com/nRFCloud/application-protocols/blob/v1/schemas/deviceToCloud/device/device.json
+ */
+#define MODEM_DYNAMIC_BATCH_EXAMPLE \
+"[{"\
+	"\"appId\":\"DEVICE\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123,"\
+	"\"data\":{"\
+		"\"networkInfo\":{"\
+			"\"currentBand\":3,"\
+			"\"networkMode\":\"NB-IoT\","\
+			"\"rsrp\":-8,"\
+			"\"areaCode\":12,"\
+			"\"mccmnc\":24202,"\
+			"\"cellID\":33703719,"\
+			"\"ipAddress\":\"10.81.183.99\""\
+		"}"\
+	"}"\
+"},{"\
+	"\"appId\":\"RSRP\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123,"\
+	"\"data\":\"-8\""\
+"}]"
+
+#define SENSORS_BATCH_EXAMPLE \
+"[{"\
+	"\"appId\":\"AIR_QUAL\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123,"\
+	"\"data\":\"50\""\
+"},{"\
+	"\"appId\":\"HUMID\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123,"\
+	"\"data\":\"50.00\""\
+"},{"\
+	"\"appId\":\"TEMP\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123,"\
+	"\"data\":\"23.00\""\
+"},{"\
+	"\"appId\":\"AIR_PRESS\","\
+	"\"messageType\":\"DATA\","\
+	"\"ts\":1563968747123,"\
+	"\"data\":\"101.00\""\
+"}]"
+
+const static struct cloud_data_cfg config_struct_example = {
+	.active_mode = false,
+	.gnss_timeout = 60,
+	.active_wait_timeout = 120,
+	.movement_resolution = 120,
+	.movement_timeout = 3600,
+	.accelerometer_activity_threshold = 10,
+	.accelerometer_inactivity_threshold = 5,
+	.accelerometer_inactivity_timeout = 1,
+	.no_data = {
+		.gnss = true,
+		.neighbor_cell = true,
+	},
+};
+
+void test_enc_neighbor_cells_empty(void)
```

Comment:
May be a one liner explanation would be nice for these tests. For me atleast the test case names dont fully convey what is being tested.

Reply:
done - that took quite a while